pyOCD/pyDAPAccess/analyzer/cmsis_dap_analyzer.py

```diff
@@ -0,0 +1,464 @@
+            tcnt = tcntH << 8 | tcntL
+
+            header = " (dap_idx=0x{:02x}, xfer_cnt=0x{:02x})".format(dapidx,tcnt)
+            trace = header + self._xrqst_block(tcnt, treq, xfers)
+        else:
+            tracer = self.get_rqst_translater(cmd)
+            trace = tracer(data) if tracer is not None else ""
+
+        # logging.debug(("trace_tx:%s" % cmd.lower()) + trace)
+
+    def trace_read(self, data, **kwargs):
+        cmd = lookup_cmd(data[0])
+        if cmd == 'DAP_TRANSFER':
+            tcnt, tresp, xfers = data[1], data[2]&0x1F, data[3:]
+            swderr = (tresp&0b01000) >> 3
+            vmm    = (tresp&0b10000) >> 4
+
+            header = " (xfer_cnt=0x{:02x}, resp={:s}, swderr={:d}, vmismatch={:d})"
+            header = header.format(tcnt,lookup_code(TRANSFER_RESP,tresp),swderr,vmm)
+            trace = header + self._xresp(tcnt, tresp, xfers)
+        elif cmd == 'DAP_TRANSFER_BLOCK':
+            (tcntL,tcntH), tresp, xfers = data[1:3], data[3]&0xF, data[4:]
+            tcnt = tcntH << 8 | tcntL
+            swderr = (tresp&0b01000) >> 3
+
+            header = " (xfer_cnt=0x{:02x}, resp={:s}, swderr={:d})"
+            header = header.format(tcnt,lookup_code(TRANSFER_RESP,tresp),swderr)
+            trace = header + self._xresp_block(tcnt, tresp, xfers)
+        else:
+            tracer = self.get_resp_translater(cmd)
+            trace = tracer(data) if tracer is not None else ""
+
+        logging.debug(("trace_rx:%s" % cmd.lower()) + trace)
+        
+    def dap_response(self, data):
+        return " (resp=%s)" % lookup_code(RESP_STATUS, data[1])
+
+    def _xrqst(self, tcnt, xfers):
+        # for each block transfer, a single transfer request byte is sent that
+        # applies to all data sent/received
+        lines = ''
+        for i in range(tcnt):
+            lines += '\n\t'
+            treq = xfers.pop(0)&0x3F
+
+            apndp = (treq&0b000001)
+            rnw =   (treq&0b000010) >> 1
+            a23 =   (treq&0b001100)
+            vm =    (treq&0b010000) >> 4
+            mm =    (treq&0b100000) >> 5
+
+            dapregmap = AP_REG if apndp else DP_RREG if rnw else DP_WREG
+            dapreg = ('AP' if apndp else 'DP') + '.' + dapregmap[a23]
+
+            data = None
+            lines += ('READ ' if rnw else 'WRITE') + ' ' + dapreg
+            if not rnw:
+                dword = get_word(xfers, pop=True)
+                lines += ", 0x{:08X}".format(dword)
+                data = dword
+
+                if dapregmap[a23] == 'TAR':
+                    lines += " ;  TAR = &%s" % REG_MAP.get(dword,"0x{:08X}".format(dword))
+                elif dapregmap[a23] == 'DRW':
+                    lines += " ; *TAR = 0x{:08X}".format(dword)
+            elif dapregmap[a23] == 'DRW':
+                lines += " "*12 + " ;  DRW = *TAR"
+
+            if dapregmap[a23] == 'TAR':
+                regs = sorted([(a,b) for (a,b) in MEM_MAP if a <= dword < b])
```

Comment:
I get an exception here because 'dword' is undefined:
```
UnboundLocalError: local variable 'dword' referenced before assignment
```

The command I used to cause this is:
```
pyocd-gdbserver -d trace
```